repose-aggregator/components/filters/ip-classification/src/main/resources/META-INF/schema/config/ip-classification.xsd

```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+  Repose
+  _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+  Copyright (C) 2010 - 2015 Rackspace US, Inc.
+  _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+  -->
+
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
+           xmlns:html="http://www.w3.org/1999/xhtml"
+           xmlns:vc="http://www.w3.org/2007/XMLSchema-versioning"
+           xmlns:xerces="http://xerces.apache.org"
+           xmlns:saxon="http://saxon.sf.net/"
+           xmlns="http://docs.openrepose.org/repose/ip-classification/v1.0"
+           targetNamespace="http://docs.openrepose.org/repose/ip-classification/v1.0"
+           elementFormDefault="qualified"
+           attributeFormDefault="unqualified">
+
+    <xs:element name="ip-classification" type="IpClassificationConfig"/>
+
+    <xs:complexType name="IpClassificationConfig">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>The root configuration for the Repose IP Classification filter configuration file.</html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:all>
+            <xs:element name="user-header-name" type="HeaderType" minOccurs="0" maxOccurs="1" default="x-pp-user"/>
+            <xs:element name="group-header-name" type="HeaderType" minOccurs="0" maxOccurs="1" default="x-pp-groups"/>
+            <xs:element name="classifications" type="ClassificationsType" minOccurs="1" maxOccurs="1"/>
+        </xs:all>
+    </xs:complexType>
+
+    <!-- because this becaomse a complex type now, I cannot use defaults. Have to put them in the scala :( -->
+    <!-- well, the quality could be optional, but not the name, and if the name isn't there, the quality isn't there either -->
```

Comment:
Yeah. I've covered it in code, but it's annoying. I prefer to go on the side of clearer configuration, and we'll deal with the complexity in the code, because we're putting this in front of users.

What do you think is best for the user?